mbed-trace/mbed_trace.h

```diff
@@ -110,7 +110,21 @@ extern "C" {
 #define tr_cmdline(...)         mbed_tracef(TRACE_LEVEL_CMD,     TRACE_GROUP, __VA_ARGS__)   //!< Special print for cmdline. See more from TRACE_LEVEL_CMD -level
 
 /** Possible to skip all traces in compile time */
-#if defined(YOTTA_CFG_MBED_TRACE)
+#if defined(YOTTA_CFG_MBED_TRACE) || (defined(YOTTA_CFG) && !defined(NDEBUG))
+
+/**
+ * Allow specification of default TRACE_GROUP to be used if not specified by application
+ */
+
+#ifndef TRACE_GROUP
+#ifdef YOTTA_CFG_MBED_TRACE_GROUP
+#define STR_HELPER(x) #x
+#define STR(x) STR_HELPER(x)
```

Comment:
Yes, I agree.
This is public header file, so you are polluting public namespace with macros `STR` and `STR_HELPER` which are too generic, I think.

Refactor those to be `TRACE_GROUP_STR` and `TRACE_GROUP_STR_HELPER`